src/pluginlist.cpp

```diff
@@ -182,82 +182,85 @@ void PluginList::refresh(const QString& profileName,
-      bool forceDisabled =
-          m_GamePlugin->loadOrderMechanism() == IPluginGame::LoadOrderMechanism::None &&
-          !forceLoaded && !forceEnabled;
-      if (!lightPluginsAreSupported && filename.endsWith(".esl")) {
-        forceDisabled = true;
-      }
+    bool archive = false;
+    try {
+      FilesOrigin& origin = baseDirectory.getOriginByID(current->getOrigin(archive));
 
-      bool archive = false;
-      try {
-        FilesOrigin& origin = baseDirectory.getOriginByID(current->getOrigin(archive));
-
-        // name without extension
-        QString baseName = QFileInfo(filename).completeBaseName();
-
-        QString iniPath = baseName + ".ini";
-        bool hasIni     = baseDirectory.findFile(ToWString(iniPath)).get() != nullptr;
-        std::set<QString> loadedArchives;
-        QString candidateName;
-        for (FileEntryPtr archiveCandidate : files) {
-          candidateName = ToQString(archiveCandidate->getName());
-          if (candidateName.startsWith(baseName, Qt::CaseInsensitive) &&
-              (candidateName.endsWith(".bsa", Qt::CaseInsensitive) ||
-               candidateName.endsWith(".ba2", Qt::CaseInsensitive))) {
-            loadedArchives.insert(candidateName);
-          }
-        }
+      // name without extension
+      QString baseName = QFileInfo(filename).completeBaseName();
 
-        QString originName    = ToQString(origin.getName());
-        unsigned int modIndex = ModInfo::getIndex(originName);
-        if (modIndex != UINT_MAX) {
-          ModInfo::Ptr modInfo = ModInfo::getByIndex(modIndex);
-          originName           = modInfo->name();
+      QString iniPath = baseName + ".ini";
+      bool hasIni     = baseDirectory.findFile(ToWString(iniPath)).get() != nullptr;
+      std::set<QString> loadedArchives;
+      for (const auto& archiveName : archiveCandidates) {
+        if (archiveName.startsWith(baseName, Qt::CaseInsensitive)) {
+          loadedArchives.insert(archiveName);
         }
+      }
 
-        m_ESPs.push_back(ESPInfo(filename, forceLoaded, forceEnabled, forceDisabled,
-                                 originName, ToQString(current->getFullPath()), hasIni,
-                                 loadedArchives, lightPluginsAreSupported,
-                                 overridePluginsAreSupported));
-        m_ESPs.rbegin()->priority = -1;
-      } catch (const std::exception& e) {
-        reportError(
-            tr("failed to update esp info for file %1 (source id: %2), error: %3")
-                .arg(filename)
-                .arg(current->getOrigin(archive))
-                .arg(e.what()));
+      QString originName    = ToQString(origin.getName());
+      unsigned int modIndex = ModInfo::getIndex(originName);
+      if (modIndex != UINT_MAX) {
+        ModInfo::Ptr modInfo = ModInfo::getByIndex(modIndex);
+        originName           = modInfo->name();
       }
+
+      m_ESPs.emplace_back(ESPInfo(filename, forceLoaded, forceEnabled, forceDisabled,
```

Comment:
This change from `push_back` to `emplace_back` is useless if you still construct `EPSInfo` here.